Desktop: Accessibility: Add setting to increase scrollbar and other small control sizes

personalizedrefrigerator opened the pull request:
# Summary

**Goal**: Allow users to make scrollbars and checkboxes meet [WCAG 2.2 SC 2.5.8: Content Size (Minimum)](https://www.w3.org/TR/WCAG22/#target-size-minimum).

This pull request adds a setting that improves the accessibility of two controls: scrollbars and sidebar checkboxes. Both are restyled to have a minimum dimension of `24px`.

**Note**: An alternate approach &mdash; a setting that switches to system scrollbars is [implemented here.](https://github.com/laurent22/joplin/compare/dev...personalizedrefrigerator:joplin:pr/accessibility/scrollbar-accessibility) However, I haven't found a simple way to increase the size of the OS scrollbars used in Joplin from system settings. For example, this [Windows thread links to an article that suggests editing the registry](https://answers.microsoft.com/en-us/windows/forum/all/how-do-i-widen-the-scroll-bar-in-windows-11/1060ee4f-d8f1-4c32-bb06-1b927b0da45c).

# Screenshots

**The setting**:
![checkbox checked \line Larger controls \line Increases the size of scrollbars and otherwise-small controls. This makes scrollbars and checkboxes easier to click.](https://github.com/user-attachments/assets/34fcb836-ab24-40d9-a649-9baa9e62ec46)


**Sidebar, note list, Markdown editor and viewer visible** (setting enabled):
![a long note is open in the editor, the scrollbar is roughly four characters wide](https://github.com/user-attachments/assets/a9409260-d895-4a87-a648-734537fa4966)

**Rich Text Editor** (setting enabled):
![the Rich Text Editor's scrollbar is also resized](https://github.com/user-attachments/assets/27f3f2a7-f0b2-4449-bfa9-dd9ad2215bea)

**Revision viewer** (setting enabled):
![image](https://github.com/user-attachments/assets/15ff0789-fb08-44d6-a1a1-fdb459d643bd)


# Notes

- **Changing the scrollbar styles**: The `StyleSheetContainer` component was renamed to `AppStyles`. In addition to adding style sheets to each Joplin window, the component now adds a `-large-controls` CSS class to the `:root` element when the large controls setting is enabled.
    - The `.-large-controls` class is used in `main.scss` to set the scrollbar and default checkbox size.
    - Scrollbar styling is done by `packages/renderer` for rendered Markdown.
- **Refactoring changes**: The revision viewer was refactored to use the same `useMarkupToHtml` hook for rendering as the `NoteEditor`. This change was intended to simplify passing the `increaseControlsSize` setting to the renderer.
   - To use the hook, `NoteRevisionViewer` was converted from a class to a function component.


# To-do

- [ ] (Optional) This pull request refactors parts of `packages/renderer` and `packages/app-desktop`. To these changes easier to review,  it may make sense to move them to a separate pull request.
- [ ] **Checkboxes**: Rather than increasing the visual note list checkbox size (as is done currently), it make sense to instead increase the size of the checkbox's clickable region and keep its visual size the same.
   - If this is done, the clickable size of note checkboxes would be increased both with the relevant setting enabled and disabled.


# Testing plan

**Fedora 41**:
1. Start Joplin
2. Verify that scrollbars in the note viewer, editor, Rich Text editor, and revision viewer are small.
3. Enable the "larger controls" setting.
4. Verify that scrollbars in settings, the sidebar, the note list, the Rich Text editor, the Markdown viewer, and the revision viewer are all larger.
5. Open a note in a new window.
6. Repeat step 4.
<!--

Please prefix the title with the platform you are targetting:

Here are some examples of good titles:

- Desktop: Resolves #123: Added new setting to change font
- Mobile, Desktop: Fixes #456: Fixed config screen error
- All: Resolves #777: Made synchronisation faster

And here's an explanation of the title format:

- "Desktop" for the Windows/macOS/Linux app (Electron app)
- "Mobile" for the mobile app (or "Android" / "iOS" if the pull request only applies to one of the mobile platforms)
- "CLI" for the CLI app

If it's two platforms, separate them with commas - "Desktop, Mobile" or if it's for all platforms, prefix with "All".

If it's not related to any platform (such as a translation, change to the documentation, etc.), simply don't add a platform.

Then please append the issue that you've addressed or fixed. Use "Resolves #123" for new features or improvements and "Fixes #123" for bug fixes.

AND PLEASE READ THE GUIDE: https://github.com/laurent22/joplin/blob/dev/readme/dev/index.md

-->

tomasz1986 commented:
I would like to suggest having an option to set it to `narrow` (the current version), `normal`, and `wide`. The reason is that the default scrollbars in Joplin are too narrow (thus difficult to click and see), but these are just way too huge to be useful in most cases (except for those that need such large elements due to severe eyesight or motion issues).

For example, I am shortsighted, but setting scrollbars to `10px` with custom CSS is more than enough for me to see them with no issues. The default ones, I can barely see.

personalizedrefrigerator commented:
> I would like to suggest having an option to set it to narrow (the current version), normal, and wide. The reason is that the default scrollbars in Joplin are too narrow (thus difficult to click and see), but these are just way too huge to be useful in most cases (except for those that need such large elements due to severe eyesight or motion issues).

Good idea! I've converted the setting to a dropdown and updated the screenshots in the pull request description. The medium scrollbars have a width of `12px`:
![screenshot: Notebook list, note list, note editor, and note viewer, all have scrollbars wider than the default](https://github.com/user-attachments/assets/24f4727d-2d73-431e-8df5-19b2581bc352)

personalizedrefrigerator commented on packages/renderer/noteStyle.ts:
```diff
@@ -117,9 +118,17 @@ export default function(theme: any, options: Options = null) {
 			background-color: ${theme.codeBackgroundColor};
 		}
+
+		:root {
+			--scrollbar-size: ${Number(options.scrollbarSize ?? 7)}px;
+		}
+
 		::-webkit-scrollbar {
```
Try to refactor this so that the `::-webkit-scrollbar` styles aren't duplicated. This can potentially be done in a follow-up pull request.
